visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -1,37 +1,40 @@
+import { Injectable, Inject } from "@angular/core"
 import { CCFileValidationResult, checkErrors, checkWarnings } from "./util/fileValidator"
 import { NodeDecorator } from "./util/nodeDecorator"
-import { StoreService } from "./state/store.service"
 import { setFiles, setStandardByNames } from "./state/store/files/files.actions"
 import { FileSelectionState, FileState } from "./model/files/files"
 import { getCCFile } from "./util/fileHelper"
 import { NameDataPair } from "./codeCharta.model"
-import { onStoreChanged } from "./state/angular-redux/onStoreChanged/onStoreChanged"
 import { referenceFileSelector } from "./state/selectors/referenceFile/referenceFile.selector"
 import { MatDialog } from "@angular/material/dialog"
 import { ErrorDialogComponent } from "./ui/dialogs/errorDialog/errorDialog.component"
 import { loadFilesValidationToErrorDialog } from "./util/loadFilesValidationToErrorDialog"
+import { Store } from "./state/angular-redux/store"
+import { State } from "./state/angular-redux/state"
+import { tap } from "rxjs"
 
+@Injectable({ providedIn: "root" })
 export class CodeChartaService {
 	static ROOT_NAME = "root"
 	static ROOT_PATH = `/${CodeChartaService.ROOT_NAME}`
 	static readonly CC_FILE_EXTENSION = ".cc.json"
 	private fileStates: FileState[] = []
 	private recentFiles: string[] = []
-	unsubscribeReferenceFileSubscription = onStoreChanged(referenceFileSelector, (_, newReferenceFile) => {
-		if (newReferenceFile) {
-			CodeChartaService.updateRootData(newReferenceFile.map.name)
-		}
-	})
-
-	static instance: CodeChartaService
+	referenceFileSubscription = this.store
+		.select(referenceFileSelector)
+		.pipe(
+			tap(newReferenceFile => {
+				if (newReferenceFile) {
+					CodeChartaService.updateRootData(newReferenceFile.map.name)
+				}
+			})
+		)
+		.subscribe()
```

Comment:
Is it not necessary to unsubscribe from this subscription elsewhere?

Reply:
This is a root level service which gets never destroyed.